src/thread.rs

```diff
@@ -29,29 +30,37 @@ pub enum SchedulingAction {
 
 /// A thread identifier.
 #[derive(Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq, Hash)]
-pub struct ThreadId(usize);
+pub struct ThreadId(u32);
 
 /// The main thread. When it terminates, the whole application terminates.
 const MAIN_THREAD: ThreadId = ThreadId(0);
 
+impl ThreadId {
+    pub fn to_u128(self) -> u128 {
+        self.0.try_into().unwrap()
```

Comment:
```suggestion
        self.0.into()
```
But why do you even go through `u128`? Our `from_uint` methods are perfectly capable for handling `u32`.